test-network-function/lifecycle/suite.go

```diff
@@ -313,16 +319,16 @@ func testPodsRecreation(env *config.TestEnvironment) {
 			closeOcSessionsByNode(env.PartnerContainers, n.Name)
 			// drain node
 			drainNode(n.Name) // should go in this
-
-			waitForAllDeploymentsReady(env.NameSpaceUnderTest, scalingTimeout, scalingPollingPeriod)
-
-			// verify deployments are ready again
-			_, notReadyDeployments = getDeployments(env.NameSpaceUnderTest)
-			if len(notReadyDeployments) != 0 {
-				uncordonNode(n.Name)
-				ginkgo.Fail(fmt.Sprintf("did not create replicas when node %s is drained", n.Name))
+			for _, ns := range env.NameSpacesUnderTest {
+				waitForAllDeploymentsReady(ns, scalingTimeout, scalingPollingPeriod)
```

Comment:
I wonder if we should check all namespaces. Or at least the ocp namespaces when draining master node.

Reply:
can we limit the drain to worker node only ? In real deployment master node should not have any deployment under test on it

Reply:
That was the next thought I had too after leaving the comment :)
Then SNO and 3 node clusters came to my mind ...
We would definitely skip intrusive tests on SNO. 3 node cluster though seems to be a real thing for some edge scenarios.

Reply:
If the current implementation works reliably on our 3 node cluster, mark this one as resolved